### PR TITLE
Added vulnerabilities to rec flow.

### PR DESCRIPTION
**Pull request by bills7281**
image

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/PartATemplateReplacementService.kt [hunk 1]
```diff
@@ -51,15 +71,20 @@ internal class PartATemplateReplacementService {
...
 
     mappings.putAll(convertToSelectedAlternativesMap(partAData.selectedAlternatives))
     mappings.putAll(convertToSelectedStandardConditionsBreachedMap(partAData.selectedStandardConditionsBreached))
+    mappings.putAll(convertToSelectedVulnerabilitiesMap(partAData.vulnerabilities))
 
     return mappings
   }
 
+  private fun hasVulnerabilities(partAData: PartAData) =
+    partAData.vulnerabilities?.selected?.isNotEmpty() == true && !partAData.vulnerabilities.selected.any { it.value == NONE.name }
```
Comment (jonnywyatt) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/PartATemplateReplacementService.kt:86]: what if the value is `UNKNOWN`?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/integration/requests/makerecalldecisions/UpdateRecommendationRequest.kt [hunk 2]
```diff
@@ -4,135 +4,166 @@ import uk.gov.justice.digital.hmpps.makerecalldecisionapi.jpa.entity.Status
...
-			}
-		]
-	},
-	"localPoliceContact": {
-		"contactName": "Thomas Magnum",
-		"phoneNumber": "555-0100",
-		"faxNumber": "555-0199",
-		"emailAddress": "thomas.magnum@gmail.com"
-	}
+  "custodyStatus": {
```
Comment (jonnywyatt) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/integration/requests/makerecalldecisions/UpdateRecommendationRequest.kt:7]: Maybe you and Jack have different spacing set up (tabs / spaces), could be worth making them consistent?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/PartATemplateReplacementServiceTest.kt [hunk 3]
```diff
@@ -160,6 +225,7 @@ internal class PartATemplateReplacementServiceTest : ServiceTestBase() {
       assertThat(result["phone_number"]).isEqualTo("555-0100")
       assertThat(result["fax_number"]).isEqualTo("555-0199")
       assertThat(result["email_address"]).isEqualTo("thomas.magnum@gmail.com")
+      assertThat(result["email_address"]).isEqualTo("thomas.magnum@gmail.com") // TODO - BS how do I handle checkboxes
```
Comment (jonnywyatt) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/PartATemplateReplacementServiceTest.kt:228]: Is the duplicate assertion here intended?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/integration/requests/makerecalldecisions/UpdateRecommendationRequest.kt (hunk 2 above)
Comment (bills7281) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/integration/requests/makerecalldecisions/UpdateRecommendationRequest.kt:7]: IntelliJ doesn't json format strings AFAIK.
I formatted this in my browser on the json lint site!
I'll check with Jack when he gets back what he does and copy that in future to be consistent.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/PartATemplateReplacementServiceTest.kt (hunk 3 above)
Comment (bills7281) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/PartATemplateReplacementServiceTest.kt]: removed :)

File: src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/PartATemplateReplacementService.kt (hunk 1 above)
Comment (bills7281) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/PartATemplateReplacementService.kt]: Well spotted! I've corrected the logic.

**Review by jonnywyatt — APPROVED** [review]

**Pull request merged**